src/rgw/driver/rados/rgw_rest_user.cc

```diff
@@ -111,7 +113,15 @@ void RGWOp_User_Info::execute(optional_yield y)
   op_state.set_fetch_stats(fetch_stats);
   op_state.set_sync_stats(sync_stats);
 
-  op_ret = RGWUserAdminOp_User::info(s, driver, op_state, flusher, y);
+  // dump_keys is false if users-without-keys is 'read' and
+  // the user is not the system user
+  int keys_perm = s->user->get_info().caps.check_cap("users-without-keys", RGW_CAP_READ);
```

Comment:
Any specific reason to not returning the error in case of RGW_CAP_WRITE?

Reply:
The intention is that there's no point in setting the cap to `write` because it wouldn't allow any operations from happening that were already not allowed. Setting the cap to `write` just means it's not set to `read`, and the `read` bit is the only bit we care about.

Making a config variable variable that just hid the keys crossed my mind, but I needed something that could be applied to a particular user, and would be uniform across all RGWs. A new cap made the most sense.